pkg/api/types.go

```diff
@@ -44,6 +44,7 @@ type Properties struct {
 	AgentPoolProfiles       []AgentPoolProfile      `json:"agentPoolProfiles"`
 	LinuxProfile            LinuxProfile            `json:"linuxProfile"`
 	WindowsProfile          WindowsProfile          `json:"windowsProfile"`
+	ExtensionsProfile       []ExtensionProfile      `json:"extensionsProfile"`
```

Comment:
If this is targeting only Kubernetes for now please add a check for this (and it can be removed later when other orchestrators are supported).